packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -199,13 +203,148 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
         });
     }
 
+    /**
+     * @param filePath Path of the file.
+     * @param rootUris URIs of the root folders in the current workspace.
+     * @returns URI of the root folder that a file is belong to.
+     * If a file belongs to more than one root folder, return the root folder closest to the file.
+     * If the file is not from the current workspace, return an empty URI.
+     */
+    getRoot(filePath: string, rootUris: string[]): URI {
+        const roots = rootUris.filter(root => new URI(root).withScheme('file').isEqualOrParent(new URI(filePath).withScheme('file')));
+        if (roots.length > 0) {
+            return new URI(roots.sort((r1, r2) => r2.length - r1.length)[0]);
+        }
+        return new URI();
+    }
```

Comment:
We can replace this method with the following method from the `WorkspaceService`:

https://github.com/eclipse-theia/theia/blob/137840a63cb3a250d11a575fee65a294ef6a4e0b/packages/workspace/src/browser/workspace-service.ts#L599-L604